server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -849,4 +838,27 @@ private boolean secretOwnerNotProvided(String secretOwner) {
   private boolean shouldInferSecretOwnerUponCreation() {
     return config.getNewSecretOwnershipStrategy() == KeywhizConfig.NewSecretOwnershipStrategy.INFER_FROM_CLIENT;
   }
+
+  private String findClientGroup(AutomationClient automationClient) {
+      Set<Group> clientGroups = aclDAO.getGroupsFor(automationClient);
+      if (clientGroups.size() == 0) {
+        logger.warn(String.format("Client %s does not belong to any group.", automationClient));
+      } else if (clientGroups.size() == 1) {
+        Group clientGroup = clientGroups.stream().findFirst().get();
+        return clientGroup.getName();
+      } else {
+        String groups = clientGroups.stream().map(group -> group.getName()).collect(Collectors.joining(","));
+        logger.warn(String.format("Client %s belongs to more than one group: %s",
+            automationClient,
+            groups));
+      }
```

Comment:
If you collect the names into any kind of Collection (Collection<String> or List<String>), you can take advantage of the default toString() from AbstractCollection:

https://docs.oracle.com/javase/8/docs/api/java/util/AbstractCollection.html#toString--